java/ql/src/Security/CWE/CWE-918/RequestForgery.qll

```diff
@@ -0,0 +1,70 @@
+import java
+import semmle.code.java.dataflow.FlowSources
+import semmle.code.java.frameworks.javase.URI
+import semmle.code.java.frameworks.javase.URL
+import semmle.code.java.frameworks.javase.Http
+import semmle.code.java.frameworks.spring.Spring
+import semmle.code.java.dataflow.DataFlow
+
+module RequestForgery {
+  import RequestForgeryCustomizations::RequestForgery
+
+  /**
+   * A taint-tracking configuration for reasoning about request forgery.
+   */
+  class RequestForgeryRemoteConfiguration extends TaintTracking::Configuration {
+    RequestForgeryRemoteConfiguration() { this = "Server Side Request Forgery" }
+
+    override predicate isSource(DataFlow::Node source) { source instanceof RemoteFlowSource }
+
+    override predicate isSink(DataFlow::Node sink) { sink instanceof Sink }
+
+    override predicate isAdditionalTaintStep(DataFlow::Node pred, DataFlow::Node succ) {
+      additionalStep(pred, succ)
+    }
+  }
+
+  class RequestForgeryLocalConfiguration extends TaintTracking::Configuration {
```

Comment:
In production code this probably happens quite a bit. In prod apps, you may have a configurable hostname that is read from a config file. I feel like this would most often cause false positives.

If you're already local on the machine, SSRF may be the least of your problems.

Reply:
I just added this as all other queries add this.